api/db/v1/mongodb_types.go

```diff
@@ -0,0 +1,39 @@
+package v1
+
+import (
+	kommonsv1 "github.com/flanksource/kommons/api/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+
+type MongoDB struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              MongoDBSpec   `json:"spec,omitempty"`
+	Status            MongoDBStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+type MongoDBList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []MongoDB `json:"items"`
+}
+
+type MongoDBSpec struct {
+	PodResources `json:",inline"`
+	Storage      Storage `json:"storage,omitempty"`
+	Database     string  `json:"database,omitempty"`
+	User         string  `json:"user,omitempty"`
+	// Deprecated
```

Comment:
Deprecated is mostly for compatibility - given this has never been deployed anywhere just deleting outright would be better?

Reply:
It has been deployed through karina? This PR moves the CRD from being internal to Karina into this library

Reply:
I was not aware of that, I thought this was a new feature